plugins/modules/ec2_spot_instance.py

```diff
@@ -0,0 +1,359 @@
+                "UpdateTime": "2021-07-21T18:33:47+00:00"
+            },
+            "Type": "one-time"
+        }
+
+cancelled_spot_request_info:
+    description: The spot instance request details that has been cancelled
+    returned: always
+    type: list
+    sample: [
+            {
+                "SpotInstanceRequestId": "sir-qph6aytk",
+                "State": "cancelled"
+            },
+            {
+                "SpotInstanceRequestId": "sir-d8468pbj",
+                "State": "cancelled"
+            }
+        ]
+'''
+import time
+import datetime
+
+try:
+    import botocore
+except ImportError:
+    pass  # Taken care of by AnsibleAWSModule
+from ..module_utils.core import AnsibleAWSModule
+from ..module_utils.ec2 import AWSRetry
+from ansible.module_utils.common.dict_transformations import snake_dict_to_camel_dict
+
+
+def request_spot_instances(module, connection):
+    launch_specification = module.params.get('launch_specification')
+    launch_specification = snake_dict_to_camel_dict(launch_specification, capitalize_first=True)
+    params = dict()
+    params['LaunchSpecification'] = launch_specification
+    params['AvailabilityZoneGroup'] = module.params.get('zone_group')
+    params['InstanceCount'] = module.params.get('count')
+    params['LaunchGroup'] = module.params.get('launch_group')
+    params['SpotPrice'] = module.params.get('spot_price')
+    params['Type'] = module.params.get('spot_type')
+    # params['ValidFrom'] = module.params.get('valid_from')
+    # params['ValidUntil'] = module.params.get('valid_until')
+    params['InstanceInterruptionBehavior'] = module.params.get('interruption')
+    params['TagSpecifications'] = module.params.get('tags')
+    try:
+        request_spot_instance_response = connection.request_spot_instances(**params)
+        return request_spot_instance_response
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg='Error while creating the spot instance request')
+
+
+def cancel_spot_instance_requests(module, connection):
+    spot_instance_request_ids = module.params.get('spot_instance_request_ids')
+    params = dict()
+    params['SpotInstanceRequestIds'] = spot_instance_request_ids
+    try:
+        cancel_spot_instance_request_response = connection.cancel_spot_instance_requests(**params)
+        return cancel_spot_instance_request_response
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg='Error while cancelling the spot instance request')
+
+
+def main():
+    argument_spec = dict(
+        zone_group=dict(type='str', default='  '),
+        count=dict(type='int', default=1),
+        interruption=dict(type='str', default="terminate"),
+        launch_group=dict(type='str', default='  '),
+        launch_specification=dict(type='dict', default=dict()),
```

Comment:
When you document this key, it can be helpful for readability to put the suboptions in a separate var like this https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/ec2_ami.py#L703